lib/attestation.rb

```diff
@@ -0,0 +1,30 @@
+# typed: true
+# frozen_string_literal: true
+
+require "json"
+require "utils/popen"
+
+module Homebrew
+  module Attestation
+    COMMAND_ERROR = :command_error
+    def self.check_attestation(bottle, signing_repo)
+      cmd = ["gh", "attestation", "verify", bottle.cached_download, "-R", signing_repo, "--format", "json"]
+      begin
+        output = Utils.safe_popen_read *cmd
+      rescue ErrorDuringExecution => e
+        # TODO(joesweeney): Don't return a Hash. Use a real type or exceptions.
+        return { verified: false, error: COMMAND_ERROR, message: "Command failed with status #{e}" }
+      end
+
+      begin
+        json_output = JSON.parse(output)
+      rescue JSON::ParserError => e
+        # TODO(joesweeney): Don't return a Hash.
+        return { verified: false, error: JSON::ParserError, message: "Failed to parse JSON: #{e.message}" }
+      end
+      is_verified = json_output.length.positive?
```

Comment:
CC @josephsweeney: this seems a little weird to me -- I _think_ we just want to consider this verified by virtue of `gh attestation verify` succeeding, not as a function of the JSON, right?

Reply:
I had added this when I was getting a result where `gh attestation verify` had a zero exit code, but an empty json. In that case I think we wanted to fail, but honestly I can't figure out how to repro it. It may have been due to using an old version of `gh` though.

Reply:
I think when fixing the error handling, we can just add a check that the json is not empty and error in that case. I did observe it so we should probably handle it.

Reply:
Makes sense to me. IMO `json.empty?` would make this a bit clearer, I'll refactor.